pkg/btf/btf.go

```diff
@@ -87,6 +90,57 @@ func NewBTF() (*btf.Spec, error) {
 	return btf.LoadSpec(btfFile)
 }
 
+func AddModulesToSpec(spec *btf.Spec, kmods []string) (*btf.Spec, error) {
+	allTypes := []btf.Type{}
+	modulePaths := []string{}
+
+	iter := spec.Iterate()
+	for iter.Next() {
+		allTypes = append(allTypes, iter.Type)
+	}
+
+	for _, module := range kmods {
+		path := filepath.Join("/sys/kernel/btf", module)
+		f, err := os.Open(path)
+		if err != nil {
+			logger.GetLogger().Warnf("btf: Path %s does not exist", path)
```

Comment:
nit: It's more idiomatic to use structured logging: logger.GetLogger().WithError(err).WithField("path", path).Warnf(...)

Reply:
Thanks, fixed!